packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -260,6 +260,8 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{
                   dataMax = dataMax > 0 ? 1 : 0;
                   return [dataMin, dataMax];
                 }
+              : config.unit === 'enumstr'
```

Comment:
🤔 This feels perhaps too much

I like the idea of adding an `enum` to https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/dataFrame.ts#L11

but then the real question is what the underlying type should be (number|string) -- seems like keeping it as number, but adding mappings for that field make the most sense